### ISSUE TITLE
could not apply CRDs

### ISSUE THREAD

**Issue by deitch** [gardener/machine-controller-manager#600]
**What happened**:

I tried to apply the CRDs, in order to test individual providers. I got a warning and an error:

```
$ kubectl apply -f kubernetes/crds/
Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
customresourcedefinition.apiextensions.k8s.io/alicloudmachineclasses.machine.sapcloud.io created
customresourcedefinition.apiextensions.k8s.io/awsmachineclasses.machine.sapcloud.io created
customresourcedefinition.apiextensions.k8s.io/azuremachineclasses.machine.sapcloud.io created
customresourcedefinition.apiextensions.k8s.io/gcpmachineclasses.machine.sapcloud.io created
customresourcedefinition.apiextensions.k8s.io/machineclasses.machine.sapcloud.io created
customresourcedefinition.apiextensions.k8s.io/machines.machine.sapcloud.io created
customresourcedefinition.apiextensions.k8s.io/machinesets.machine.sapcloud.io created
customresourcedefinition.apiextensions.k8s.io/openstackmachineclasses.machine.sapcloud.io created
customresourcedefinition.apiextensions.k8s.io/packetmachineclasses.machine.sapcloud.io created
customresourcedefinition.apiextensions.k8s.io/scales.machine.sapcloud.io created
The CustomResourceDefinition "machinedeployments.machine.sapcloud.io" is invalid: spec.validation.openAPIV3Schema.properties[metadata]: Forbidden: must not specify anything other than name and generateName, but metadata is implicitly specified
```

**What you expected to happen**:

it would apply them without error

**How to reproduce it (as minimally and precisely as possible)**:

1. Start a cluster (I used k3d v4.4.0, which uses cluster 1.20.5)
2. `kubectl apply -f kubernetes/crds/`
3. watch errors

**Anything else we need to know**:

The docs [here](https://github.com/gardener/machine-controller-manager/blob/master/docs/usage/prerequisite.md#deploy-required-crds-and-objects) are wrong, in that they say to apply `kubernetes/crds.yaml`, but `kubernetes/crds/` now is a directory

**Environment**:

Running k3d on Docker Desktop for Mac.

**Comment by deitch** [comment]
Also running the MCM gives errors (unsurprisingly):

```
E0401 18:25:21.651179    8066 reflector.go:123] github.com/gardener/machine-controller-manager/pkg/client/informers/externalversions/factory.go:117: Failed to list *v1alpha1.MachineDeployment: the server could not find the requested resource (get machinedeployments.machine.sapcloud.io)
```

**Comment by prashanth26** [comment]
Hi @deitch ,

This is an issue @AxiomSamarth had already looked into. He had proposed a fix here - https://github.com/gardener/machine-controller-manager/pull/592. I am unsure what the status is, he can probably give more details on this.

**Comment by deitch** [comment]
Darn! And I thought I had found a new one. 😄 

Thanks

**Comment by deitch** [comment]
In the meantime, how do I test `MachineDeployment`?

**Comment by prashanth26** [comment]
Does removing this part of the CRD fix it ? - https://github.com/gardener/machine-controller-manager/pull/592/files#diff-b96da678ae66c7304b1f784d2d9e3d405a5ddcb1ba251f7c47e04aefdef17a80L62-L64

**Comment by deitch** [comment]
Indeed, it does! I still get the deprecation warning, but it accepts it.

```
Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
```

**Comment by AxiomSamarth** [comment]
> Hi @deitch ,
> 
> This is an issue @AxiomSamarth had already looked into. He had proposed a fix here - #592. I am unsure what the status is, he can probably give more details on this.

Hey @prashanth26,

As Hardik has mentioned, maybe the autogeneration of the CRDs, is reverting it. So, even if we change the CRD file in the repo, it may get updated. I am not sure how I can fix this autogeneration.

**Comment by prashanth26** [comment]
> As Hardik has mentioned, maybe the autogeneration of the CRDs, is reverting it. So, even if we change the CRD file in the repo, it may get updated. I am not sure how I can fix this autogeneration.

We might have to update the k8s version (to the latest) in our vendors to permanently fix this issue - https://github.com/gardener/machine-controller-manager/issues/402. Let's close this PR and pick this issue on priority.
